fortran/example/README.md

```diff
@@ -0,0 +1,75 @@
+# How to use spglib example file in Fortran
+
+This is the README file for spglib examples for Fortran.
+
+## Examples of using spglib in a cmake project
+
+See the local [`CMakeLists.txt`](CMakeLists.txt) file for how to use
+spglib in a cmake project, using either native cmake options or pkg-config.
+
+## Examples of linking to spglib manually
+
+To manually compile `example.f90`:
+
+```console
+gfortran example.f90 ../../lib/libspglib_f08.a ../../lib/libsymspg.a -I../../include
+```
+
+Here it is assumed that spglib library was built by
+```console
+cmake -B build -DCMAKE_INSTALL_PREFIX=. -DSPGLIB_WITH_Fortran=on -DSPGLIB_SHARED_LIBS=off -DSPGLIB_WITH_TESTS=off .
+cmake --build build
+cmake --install build
+```
+
+## Concept
+
+A crystal structure is given to a spglib function or subroutine to obtain
+symmetry information. `spg_get_dataset` is the function that returns most
+fruitful space group information. This returns the result in the fortran-type of
+`SpglibDataset`. We don't know the length of symmetry operations to be returned,
+the memory spaces are allocated dynamically, which means we have to free them.
+by `spg_free_dataset`. To tolerate tiny distortion of crystal structure,
+it has to be specified when calling symmetry search function.
+No detailed documentation about crystal structure format and APIs for Fortran.
+Please refer documentation for C. The notable difference between
+those in C and Fortran are twofold: Dimensions of arrays are inverted to
+keep data order in memory unchanged, and it is unnecessary to deallocate
+`SpglibDataset`. The following pages show the details for C.
+
+- [Crystal structure format](https://spglib.github.io/spglib/variable.html)
+- [Functions](https://spglib.github.io/spglib/api.html)
+- [Spglib dataset](https://spglib.github.io/spglib/dataset.html)
+
+```fortran
+program spglib_example
+    use, intrinsic :: iso_c_binding
+    use spglib_f08, only: SpglibDataset, spg_get_dataset
+    implicit none
+
+    ! Wurtzite structure (P6_3mc)
+    real(c_double) :: lattice(3, 3)
+    real(c_double) :: positions(3, 4)
+    integer(c_int) :: atom_types(4)
+    integer(c_int) :: num_atom = 4
+    real(c_double) :: symprec = 1e-5
+    type(SpglibDataset) :: dset
+    integer :: number
+
+    ! Transposed arrays with respect to those in C format
+    ! to keep data order in memory unchanged.
+    lattice(:, 1) = [3.111, -1.5555, 0.0]
+    lattice(:, 2) = [0.0, 2.6942050311733885, 0.0]
+    lattice(:, 3) = [0.0, 0.0, 4.988]
+    positions(:, 1) = [1.0/3, 2.0/3, 0.0]
+    positions(:, 2) = [2.0/3, 1.0/3, 0.5]
+    positions(:, 3) = [1.0/3, 2.0/3, 0.6181]
+    positions(:, 4) = [2.0/3, 1.0/3, 0.1181]
+    atom_types(:) = [1, 1, 2, 2]
+
+    ! No need to deallocate SpglibDataset manually as contrasted to C
```

Comment:
This can be tested with memcheck or valgrind or...?

Reply:
I didn't test with memcheck or valgrind, etc. As written above about your comment on `c-example/README.md` (Fortran wrapper uses allocatable arrays).

Reply:
Yes, but it might be problematic depending on who initialized, e.g. if fortran initializes an instance of the structure `dataset`, but C initializes the pointers inside it, then when fortran destroys the instance, the C initialized variables still exist. Same could happen if we resize arrays and so on. A proper C++ class can automatically avoid those though.

Reply:
This is properly treated  inside the Fortran wrapper.

https://github.com/spglib/spglib/blob/02159eef6e7349535049a43fe2272bb634c77945/fortran/spglib_f08.f90#L669